rtdb2/RtDB2Monitor.cpp

```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <string>
+#include <sys/inotify.h>
+#include <boost/filesystem.hpp>
+#include "RtDB2Monitor.h"
+#include "tprintf.hpp"
+
+#define EVENT_SIZE  ( sizeof (struct inotify_event) )
+#define BUF_LEN     ( 1024 * ( EVENT_SIZE + 16 ) )
+
+namespace fs = boost::filesystem;
+
+static std::map<std::string, RtDB2Monitor*> _monitors;
+
+RtDB2Monitor& RtDB2Monitor::monitor(std::string const &path)
+{
+    std::map<std::string, RtDB2Monitor*>::iterator it = _monitors.find(path);
+    if (it == _monitors.end())
+    {
+        //tprintf("RtDB2Monitor| Setting up new monitor for path: %s", path.c_str());
```

Comment:
I also find myself sprinkling such debug statements over the code. 

Remember the Falcons tracing demo I showed (with intrusive TRACE and TRACE_SCOPED function calls and a nice google JavaScript viewer, based on chromium-catapult)? I was thinking of playing with a non-intrusive compile-time configuration, using somehow gcc `-finstrument-functions`. Just FYI.